internal/git/gitsync_processor.go

```diff
@@ -31,60 +35,98 @@ type GitSyncProcessor struct {
 	clusterName string
 }
 
-func cloneRepo(repoPath *v1.RepositoryPath) {
+func watchRepo(ctx context.Context, repo *v1.RepositoryPath, _ /* namespace */ string) error {
+	logger := logging.FromContext(ctx)
+
 	r, err := git.Clone(memory.NewStorage(), nil, &git.CloneOptions{
-		URL: repoPath.RepoUrl,
+		URL:          repo.RepoUrl,
+		SingleBranch: true,
 	})
 	if err != nil {
-		log.Fatalf("error cloning the repository %s", err.Error())
+		logger.Errorw("error cloning the repository", "err", err)
+		return err
 	}
-	// TargetRevision can be a branch, a tag, or a commit hash
-	err = checkRevision(r, repoPath.TargetRevision)
+
+	// The revision can be a branch, a tag, or a commit hash
+	h, err := r.ResolveRevision(plumbing.Revision(repo.TargetRevision))
 	if err != nil {
-		log.Fatalf("error  checking revision for the repository %s", err.Error())
+		logger.Errorw("error resolve the revision", "revision", repo.TargetRevision, "err", err)
+		return err
+	}
 
+	// Retrieving the commit object matching the hash.
+	commit, err := r.CommitObject(*h)
+	if err != nil {
+		logger.Errorw("error checkout the commit", "hash", h.String(), "err", err)
+		return err
 	}
-}
 
-func checkRevision(r *git.Repository, revision string) error {
-	hash, err := r.ResolveRevision(plumbing.Revision(revision))
+	// Retrieve the tree from the commit.
+	tree, err := commit.Tree()
 	if err != nil {
+		logger.Errorw("error get the commit tree", "err", err)
 		return err
 	}
-	w, err := r.Worktree()
+
+	// Locate the tree with the given path.
+	tree, err = tree.Tree(repo.Path)
 	if err != nil {
+		logger.Errorw("error locate the path", "err", err)
 		return err
 	}
 
-	err = w.Checkout(&git.CheckoutOptions{
-		Hash: *hash,
+	// Read all the files under the path and apply each one respectively.
+	err = tree.Files().ForEach(func(f *object.File) error {
+		logger.Debugw("read file", "file_name", f.Name)
+		_, err = f.Contents()
+		if err != nil {
+			logger.Errorw("cannot get file content", "filename", f.Name, "err", err)
+			return err
+		}
+
+		// TODO: Apply to the resources
+		return nil
 	})
 	if err != nil {
 		return err
 	}
 
-	if isCommitSHA(revision) {
-		log.Println("No monitoring")
+	if isCommitSHA(repo.TargetRevision) {
+		// TODO: no monitoring
+		logger.Debug("no monitoring")
```

Comment:
if we're keeping these log lines, can we add more information to them?

Reply:
I don't think we will keep the log, or at least we will know what to log in next iterations when we actually add the monitoring. The log here is only to avoid lint error.